Add Blazor topics

guardrex opened the pull request:
Fixes #10825 

Internal Review Topics:

* [Blazor Overview](https://review.docs.microsoft.com/en-us/aspnet/core/client-side/spa/blazor/index?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Blazor: Get started](https://review.docs.microsoft.com/en-us/aspnet/core/client-side/spa/blazor/get-started?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Razor Components Overview](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/index?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Razor Components: Get started](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/get-started?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Build your first app](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/build-your-first-razor-components-app?view=aspnetcore-3.0&branch=pr-en-us-10873)

Notes:

* Can't name both RC and Blazor samps with the same name, so I went with "Razor Components" and "Blazor" as the app names ... super-obvious that way when readers look in the tutorial's *samples* folder.
* Still needs work on the intro paragraphs of each of the introduction (*Overview*/*index.md*) topics. Blazor intro needs the list of main features. If you need to make big layout changes, I recommend that you edit these directly here, then I'll do a style+grammar pass afterward.
* Made sense to jettison the "hosting models" sections now that these intro topics are entirely dedicated to either RC or Blazor. I retain a brief remark at the end of each about the converse approach (RC mentions Blazor and Blazor mentions RC) with a cross-link.
* Dan, your recommendations worked well. I 🖖 u on those calls.

danroth27 commented on aspnetcore/razor-components/index.md:
```diff
@@ -20,35 +20,7 @@ By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.
 * Share server-side and client-side app logic all written with .NET.
```
I think we should remove the include at the beginning of this doc that adds the note about Blazor being experimental. It's not really relevant for this topic.

danroth27 commented on aspnetcore/tutorials/build-your-first-razor-components-app.md:
```diff
@@ -143,7 +153,7 @@ Add a new page to the app that implements a simple todo list.
-   [!code-cshtml[](build-your-first-razor-components-app/samples/3.x/WebApplication1/WebApplication1.App/TodoItem.cs)]
+   [!code-cshtml[](build-your-first-razor-components-app/samples/3.x/RazorComponents/RazorComponents.App/TodoItem.cs)]
 
 1. Return to the Todo component (*Todo.cshtml*):
```
The highlighting in the code snippet for this section looks a bit off.

guardrex replied:
I double-checked locally, and it looks :eye: ok. I can't see the doc's system build (internal review topic), but I'll find out quickly right after it goes live what's what. I can do a quick patch if anything breaks. Locally, these highlights seem to be ok.

guardrex commented on aspnetcore/client-side/spa/blazor/get-started.md:
```diff
@@ -0,0 +1,163 @@
+
+By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/razor-components-preview-notice.md)]
```
Cool ... I'll make this update on the PR this afternoon that also addresses the Blazor TOC additions. I'll do it right after we merge this one.

guardrex commented:
@tdykstra You can make it a quickie look (I'd like to merge and get right back into these topics with an updated master) ... we'll be back here for Preview 3 making a final pass. I'm mostly concerned about your approval of the TOC bits.

@scottaddie We'll have time for u to :eye: it for Preview 3 ..... you should be dug out ⛄️ by then I hope.

danroth27 commented:
@tdykstra These TOC updates sound fine to me, although I would prefer to have Razor Components in first position, and then SPA second under client-side dev.

guardrex commented:
> A bunch of TOC nodes under Razor Components 404 -- is that expected at this point?

I'm not getting that here. No RC topics 404 here. Which ones are 404'ing there?

> delete Overview under Single Page Apps (it 404s)

It loads here as https://docs.microsoft.com/aspnet/core/client-side/spa/

I'll work a commit along the suggestions provided.

guardrex commented:
@tdykstra @danroth27 

* Pointed the Client-side Overview to the Razor Components Overview. You said Blazor, but RC is the supported tech at this point. Blazor is the first link tho under SPA. I perhaps guessed wrong, so let me know.
* Moved RC to the top of the node. 
* Moved LibMan under RC.

tdykstra commented:
>No RC topics 404 here. Which ones are 404'ing there?

They're working now.

>It loads here as https://docs.microsoft.com/aspnet/core/client-side/spa/

That one works now too.  It's one of those duplicate-the-toc files, how about delete and redirect to Blazor since we don't have a real SPA overview?

guardrex commented:
@tdykstra Done on the last commit ... see how that looks.

guardrex commented:
... but I only deleted. If you still want "Overview" but linked to the Blazor Overview, I still need to do that.

guardrex commented:
Feel free to edit directly, of course for what you want.

tdykstra commented:
OK, editing TOC now, a couple of tweaks.  Also, the /aspnet/core/client-side/spa/index shows the page now but is disconnected from the TOC.  Need to redirect the URL to Blazor

guardrex commented:
Yes, :+1: redirect entry. Let me know if u want me to add that or if you're going to do it.

guardrex commented:
@tdykstra Try that redirect ... see if that looks good. :point_up:

tdykstra commented:
Thanks! -- if the TOC changes build OK I think we're good.

guardrex commented:
Looks like we need a client-side/index redirect, too. I'll add that next.

guardrex commented:
That might&dagger; work.

**&dagger;might** ... if the :pray: *server gods* :pray: are happy with us.

guardrex commented:
@tdykstra Look at it this way .......

...... if this all goes to heck in a handbasket ........

**I hear Apple docs _ARE HIRING!_** 😀😀😀😀🙈:smile:

@danroth27 ... look good to you? ... I'll get this merged and on to the Blazor TOC/INCLUDE updates next.

guardrex commented:
It wasn't too pleased with the file left on the repo, so I 🔪 the file on the last commit.

scottaddie commented:
I agree that LibMan should appear in the ToC before Gulp and Grunt. We do own that product, after all.

danroth27 commented:
:shipit:
